src/app/content/hooks/registerPageView.ts

```diff
@@ -0,0 +1,27 @@
+import googleAnalyticsClient from '../../../gateways/googleAnalyticsClient';
+import * as selectNavigation from '../../navigation/selectors';
+import { RouteHookBody } from '../../navigation/types';
+import { AppServices, MiddlewareAPI } from '../../types';
+import { content } from '../routes';
+
+export const hookBody: RouteHookBody<typeof content> = (services: MiddlewareAPI & AppServices) => {
+  let lastTrackedLocation: any;
```

Comment:
I implemented the changes you proposed and everything looks good, except that I'm not sure how to mock this for testing. Currently `pathOrModalChanged` always evaluates to true in testing because lastTrackedLocation always starts out undefined.

Reply:
if you mean for a unit test for this hook specifically you can do something like 

```
const theHook = registerPageViewHook(mockServices)

theHook(mockLocationAction)
expect(pageviewtobetracked)
theHook(mockLocationActionThatShouldntBeTracked)
expect(pageviewnottracked)
theHook(mockLocationActionThatShouldBeTracked)
pageviewtobetracked
```

does that makes sense?

Reply:
Hmm ok I had already tried that but I think I left out the first `expect`. What's weird is when I call the hook with any action I get a tracked page view, even when the action fails the `action.action !== 'REPLACE' && pathOrModalChanged` condition.

Reply:
I pushed the spec so you can see.

Reply:
@jivey maybe you will have thoughts about this failing spec

Reply:
@bethshook I think this is due to some odd behavior on Jest's part. The GA client calls `.ga` twice here: https://github.com/openstax/rex-web/blob/1d43b5861dc02e6091a97706bba393b3b89262c6/src/gateways/googleAnalyticsClient.ts#L196 for each `trackPageView` call, and `toHaveBeenCalledWith/toHaveBeenCalledTimes` just looks at the `.mock.calls` array to decide if multiple calls were made, even though one call results in this:

```
[
  [ 'tfoo.set', 'queueTime', 0 ],
  [ 'tfoo.send', { hitType: 'pageview', page: '/' } ]
]
```

This was a little surprising to me, I always thought Jest tracked call results more granularly. Anyway, I think the fix is a mockReset between, this passes for me:

```
await hook(action);
expect(mockGa).toHaveBeenCalledWith('tfoo.send', {hitType: 'pageview', page: '/'});
mockGa.mockReset();
await hook(action);
expect(mockGa).not.toHaveBeenCalledWith('tfoo.send', {hitType: 'pageview', page: '/'});
```

Reply:
Thanks @jivey - there is always some jest idiosyncrasy that hasn't occurred to me